app/src/main/java/com/locoquest/app/dao/IBenchmarkDAO.kt

```diff
@@ -14,29 +14,39 @@ interface IBenchmarkDAO {
     fun getBenchmarkByPid(@Query("pid") pid: String): Call<Benchmark>
 }
 
-fun main() {
-    val retrofit = Retrofit.Builder()
-        .baseUrl("https://geodesy.noaa.gov/api/nde/")
+fun createRetrofitInstance(baseUrl: String): Retrofit {
+    return Retrofit.Builder()
+        .baseUrl(baseUrl)
         .addConverterFactory(MoshiConverterFactory.create())
         .build()
-
+}
+fun main() {
+    val retrofit = createRetrofitInstance("https://geodesy.noaa.gov/api/nde/")
     val benchmarkDAO = retrofit.create(IBenchmarkDAO::class.java)
 
     val pid = "AB1234"
 
     val call = benchmarkDAO.getBenchmarkByPid(pid)
-    call.enqueue(object : Callback<Benchmark> {
-        override fun onResponse(call: Call<Benchmark>, response: Response<Benchmark>) {
-            if (response.isSuccessful) {
-                val benchmark = response.body()
-                println(benchmark)
-            } else {
-                println("Error: ${response.code()}")
+    try {
+        call.enqueue(object : Callback<Benchmark> {
+            override fun onResponse(call: Call<Benchmark>, response: Response<Benchmark>) {
+                if (response.isSuccessful) {
+                    val benchmark = response.body()
+                    if (benchmark != null) {
+                        println(benchmark)
+                    } else {
+                        println("Error: Response body is null or empty")
+                    }
+                } else {
+                    println("Error: ${response.code()}")
+                }
             }
-        }
 
-        override fun onFailure(call: Call<Benchmark>, t: Throwable) {
-            println("Error: ${t.message}")
-        }
-    })
+            override fun onFailure(call: Call<Benchmark>, t: Throwable) {
+                println("Error: ${t.message}")
+            }
+        })
+    } catch (e: Exception) {
+        println("Error: ${e.message}")
+    }
```

Comment:
Added extra error handling in the OnResponse and OnFailure routes
https://elizarov.medium.com/kotlin-and-exceptions-8062f589d07

Reply:
Use proper logging instead of println()
- Logging has the concept of logging levels.  They can be filtered and aggregated based on those levels.
- No one will see println()
- 
If adding a catch block, heal the exception there.  An empty catch block is not a good option, as you're losing valuable information about why something crashed, and assuming that everything worked fine.